src/hooks/listingSupport/updateCache.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Gets an object with the correct keys/values for this concept to be cached
+ * @param {Object} concept a listing-information response from the API
+ */
+const getCacheForConcept = (concept) => {
```

Comment:
I thought it wouldn't matter whether this is a conceptId or trial type, and we are caching based on the key provided. As to that effect, believed the cache should be agnostic of what it's caching, just requires a unique key and item to be cached, and would be retrieving and updating based on key provided to it.

I'm probably missing something quite obvious here.

Reply:
Take a look at the tests to see what gets cached and what we expect the cache to be...

Reply:
I would expect a cache dump to look like it is below based on the sample cache object from test:

```javascript
{
  C1111: {
	conceptId: ['C1111', 'C2222'],
	name: {
		label: 'Multi-id Concept',
		normalized: 'multi-id concept',
	},
	prettyUrlName: 'multi-id-concept',
  },
  C2222: {
	  conceptId: ['C1111', 'C2222'],
	  name: {
		  label: 'Multi-id Concept',
		  normalized: 'multi-id concept',
	  },
	  prettyUrlName: 'multi-id-concept',
  },
  'multi-id-concept': {
	  conceptId: ['C1111', 'C2222'],
	  name: {
		  label: 'Multi-id Concept',
		  normalized: 'multi-id concept',
	  },
	  prettyUrlName: 'multi-id-concept',
  },
  treatment: {
	  prettyUrlName: 'treatment',
	  idString: 'treatment',
	  label: 'Treatment',
  },
  C99999: {
	  conceptId: ['C99999'],
	  name: {
		  label: 'No Purl Concept',
		  normalized: 'no purl concept',
	  },
	  prettyUrlName: null,
  },
}
```

Since the cache object keys are unique identifiers (in this case conceptId, prettyUrlName, or trial type), which can be different in another application (was thinking we could use this as a drop replacement in our other apps where this can be needed since all the cache should need to function would be a unique key and it's corresponding value to set /update, or just the key to retrieve.

Reply:
I am not following exactly. The function you are referencing gets the appropriate keys for those objects to cache them. And the object you pasted is the shape of the object in the tests.

Reply:
After discussing with @bryanpizzillo, I get that this file contains utility functions for listing support that interfaces with the actual cache functionality.

Reply:
In discussing with Seyi, the real issue is that listingSupportCache, listingSupportContext, listingSupportContextProvider all could be some sort of GenericKeyValuePairCache. Since the Context/Provider/Cache/Cache Fetching is really the generic thing.

However, to truly make it generic for AN entire app, it should really also have a way to "bucket" the cache items since you should only have 1 of these caches for your app. Adding buckets and renaming is outside the scope of this ticket though.